doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+`accelerated-bridge-cni` and contains the required resource requests.
+* As a user, I want to define configuration for software bridges inside the `SriovNetworkNodePolicy` CR and expect that the
+operator will create required bridges, configure them, and attach uplinks (physical functions).
+
+### Goals
+
+* handle installation of `ovs-cni` and `accelerated-bridge-cni`
+* add `OVSNetwork` and `BridgeNetwork` CRDs as an API for end users to simplify creation of `NetworkAttachmentDefinition` CR
+for `ovs-cni` and `accelerated-bridge-cni`
+* extend `SriovNetworkNodePolicy` CRD to support configuration of software bridges (bridge-level configuration)
+* extend `SriovNetworkNodeState` CRD to support configuration of software bridges (bridge-level configuration)
+* support configuration of software bridge in both modes (operator's `configurationMode` setting): `daemon` and `systemd`
+* implementation should be compatible with [_Externally Managed PF_](externally-manage-pf.md) feature
+
+
+### Non-Goals
+
+* replace `SriovNetworkPoolConfig` CRD
+* change API for host-level settings, e.g. `ovs-hw-offload`
+	
+	_**Note:** we may need to extend this API to support additional options_
+
+* add support for VF-lag use-case
+
+## Proposal
+
+1. deploy `ovs-cni` and `accelerated-bridge-cni` with init containers of `sriov-network-config-daemon` Pod
+2. define `OVSNetwork` and `BridgeNetwork` CRDs and implement controllers for them which will create `NetworkAttachmentDefinition` CRs
+3. extend `SriovNetworkNodePolicy`and `SriovNetworkNodeState` CRDs to support configuration of software bridges (bridge-level configuration)
+
+
+### Workflow Description
+
+Implementation should be compatible with the following workflows:
+
+* [Fully automatic workflow](#fully-automatic-workflow)
+* [NIC configuration only flow](#nic-configuration-only-flow) (Externally managed bridge)
+* [Externally managed NIC flow](#externally-managed-nic-flow)
+
+#### Fully automatic workflow
+
+This workflow assumes that **sriov-network-operator** handles PFs configuration, creation and configuration of software bridge, announcement of SRIOV resources with device plugin and preparation of `NetworkAttachmentDefinition` CR.
+
+1. User creates `SriovNetworkNodePolicy` where:
+    * eswitch mode set to `switchdev`
+    * configuration for selected software bridge is defined
+
+2. The operator populates `SriovNetworkNodeState` for matching nodes with PF and bridge
+configurations 
+
+3. `sriov-network-config-daemon` applies PF configuration, create and configure bridge, attach PF to the bridge, applies VF configuration
+
+
+    _**Note 1:** if the operator runs in the `systemd` mode then bridge creation should happen in the `pre` phase._
+
+    _**Note 2:** we should create udev rule which will set `NM_UNMANAGED=1` for bridges created by the operator_
+
+4. `sriov-network-config-daemon` should report information about software bridges in the status field of the `SriovNetworkNodePolicy` CR.
```

Comment:
If "fully auto" mode is used  we should create a bridge with predictable name for each PF that match a policy that contains a bridge configuration. We can't use interface name to generate a predictable name for the bridge, because the limit for the interface length is only 15 symbols and PF name will already have length around this. 
I propose to use PF's PCI address to generate predictable bridge name, like `br-0000_d8_01.0`.

Reply:
Yes, we will report this information in the status field of the state object.